src/MacroTools/Powers/Transfiguration.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using Microsoft.VisualBasic;
+using WCSharp.Events;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.Powers
+{
+  /// <summary>
+  /// Causes dying units to Transfiguring at a specific location.
+  /// </summary>
+  public sealed class Transfiguration : Power
+  {
+    private readonly float _chance;
+    private readonly Point _returnPoint;
+    private readonly Rectangle _noReturnRect;
+    public int AnimatedArmorID { get; init; } = FourCC("o01I");
+
+    /// <summary>
+    /// The condition that units need to pass to be eligible for rematerialization.
+    /// </summary>
+    public Func<unit, bool> EligibilityCondition { get; init; } = _ => true;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Transfiguring"/> class.
+    /// </summary>
+    /// <param name="chance">The chance that dying units have to rematerialize.</param>
```

Comment:
Generally: the word "rematerialize" doesn't mean anything in this context and it's used in lots of places.